### PR TITLE
Refactor sm2 add test case

### PR DESCRIPTION
**Pull request by travis-joe**

### CODE REVIEW

File: packages/cita-signer/src/unsigner.ts [hunk 1]
```diff
@@ -76,7 +75,7 @@ const unsigner = (hexUnverifiedTransaction: string) => {
     }
 
     // return result
-  } else if (citaRSA == 'sm2') {
+  } else if (cryptoTx == CryptoTx.SM2) {
```
Comment (Keith-CY) [inline packages/cita-signer/src/unsigner.ts]:
```suggestion
  } else if (cryptoTx === CryptoTx.SM2) {
```

File: packages/cita-signer/test/index.test.js [hunk 2]
```diff
@@ -1,3 +1,5 @@
+const {CryptoTx}  = require ("../lib/enum");
```
Comment (Keith-CY) [inline packages/cita-signer/test/index.test.js]:
```suggestion
const { CryptoTx }  = require ("../lib/enum");
```

File: packages/cita-signer/src/unsigner.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
-import citaRSA from './config'
+const { publicKeyToAddressSM2 } = require('../../cita-sdk/lib/utils/sm2Utils')
```
Comment (Keith-CY) [inline packages/cita-signer/src/unsigner.ts]: Does it refer to a `js` file?

Comment (travis-joe) [reply packages/cita-signer/src/unsigner.ts]: yes, so, that will be a problem?

Comment (Keith-CY) [reply packages/cita-signer/src/unsigner.ts]: I was just confused.

Now the `cita-signer` imports `cita-sdk` as its dependency while `cita-signer` is a dependency of `cita-sdk`. And the reference is not achieved by `npm package` but directly by files.

Comment (rainchen) [reply packages/cita-signer/src/unsigner.ts]: these are 2 packages,  `cita-signer` should not know there is a dir like `../../cita-sdk`
this should be fixed.

**Pull request merged**